lib/buildsystems/ruby.rb

```diff
@@ -62,9 +62,10 @@ def perform_request(req)
   gem_test = $gems.grep(/#{"^#{passed_name.gsub(/^ruby_/, '')}\\s.*$"}/).first.blank? ? $gems.grep(/#{"^#{passed_name.gsub(/^ruby_/, '').gsub('_', '-')}\\s.*$"}/).first : $gems.grep(/#{"^#{passed_name.gsub(/^ruby_/, '')}\\s.*$"}/).first
   gem_test_name = gem_test.split.first
   gem_test_versions = gem_test.split[1].split(',')
-  gem_test_versions.delete_if { |i| i.include?('beta') }
-  gem_test_versions.delete_if { |i| i.include?('pre') }
-  gem_test_version = gem_test_versions.max
+  # Any version with a letter is considered a prerelease as per
+  # https://github.com/rubygems/rubygems/blob/b5798efd348935634d4e0e2b846d4f455582db48/lib/rubygems/version.rb#L305
+  gem_test_versions.delete_if { |i| i.match?(/[a-zA-Z]/) }
```

Comment:
Do you need `+` as in `[a-zA-Z]+` or is it only looking for one character?

Reply:
If it finds ANY non-numeric character, it marks it as prerelease. That's the logic in the gem version script in rubygems...